src/core/core.tooltip.js

```diff
@@ -852,27 +852,27 @@ module.exports = function(Chart) {
 			// Remember Last Actives
 			changed = !helpers.arrayEquals(me._active, me._lastActive);
 
-			// If tooltip didn't change, do not handle the target event
-			if (!changed) {
-				return false;
-			}
+			// Only handle target event on tooltip change
+			if (changed) {
+				me._lastActive = me._active;
 
-			me._lastActive = me._active;
+				if (options.enabled || options.custom) {
+					me._eventPosition = {
+						x: e.x,
+						y: e.y
+					};
 
-			if (options.enabled || options.custom) {
-				me._eventPosition = {
-					x: e.x,
-					y: e.y
-				};
+					var model = me._model;
+					me.update(true);
 
-				var model = me._model;
-				me.update(true);
-				me.pivot();
-
-				// See if our tooltip position changed
-				changed |= (model.x !== me._model.x) || (model.y !== me._model.y);
+					// See if our tooltip position changed
+					changed |= (model.x !== me._model.x) || (model.y !== me._model.y);
+				}
 			}
 
+			// always call me.pivot before returning to
+			// reset me._start for smooth animations issue #4989
+			me.pivot();
```

Comment:
I don't think `me.pivot()` should be called if the `_active` target(s) didn't change, since it resets the animation start/end values. I think that's what caused the weird animation accelerations in the following gif.

![chartjs 5005](https://user-images.githubusercontent.com/3874900/33437747-2a37cc60-d5e9-11e7-8ae3-35670a99fac2.gif)

Reply:
@simonbrunel not calling `me.pivot()` [codepen](https://codepen.io/anon/pen/rYKQrx) causes a much weirder experience by resetting all settings to their initial values. (sorry don't have any software to make a fancy gif).
Any suggestions?

Reply:
I can't debug right now :\ Maybe `_active` is different depending on the hovered bar, so the first `changed` value is `true` because of the array order?! Though, I'm not sure why checking the latest value of `changed` doesn't work:

```javascript
  // {...}
  if (changed) {
    me.pivot();
  }

  return changed;
}
```

BTW, the following looks weird (line 869):

```javascript
if (changed) {
    // {...}
    // See if our tooltip position changed
    changed |= (model.x !== me._model.x) || (model.y !== me._model.y);
}
```

because if `changed` is `true` then `changed |= true or false` is necessary `true`, right?

Reply:
@simonbrunel I agree that line 869 is obsolete, however it was there already obsolete from a previous pull.
The problem doesn't lay in this handleEvent, but in its caller, which `|=` its `changed` with the result of the `tooltip.handleEvent` (`changed |= tooltip && tooltip.handleEvent(e);`). When the caller has changed, but the tooltip hasn't, the bug presents itself, because the animation will run. I don't understand the animation system well enough to prevent the tooltip animation restarting.

Reply:
I'm wondering if it should not be `changed &= (model.x !== me._model.x) || (model.y !== me._model.y);` instead?

Reply:
The issue seems to be that we detect a change for the [highlighted item](https://codepen.io/anon/pen/GOwVwQ) resetting the current animation. Calling `pivot()` as you do makes things better in this case but breaks animations when hover interaction mode/intersect are the same as tooltip ([master](https://codepen.io/anon/pen/NwEQme) - [PR](https://codepen.io/anon/pen/zPMgVq)).